sdk/src/main/scala/com/horizen/api/http/SidechainTransactionApiRoute.scala

```diff
@@ -447,6 +446,107 @@ case class SidechainTransactionApiRoute(override val settings: RESTApiSettings,
     }
   }
 
+  def createOpenStakeTransaction: Route = (post & path("createOpenStakeTransaction")) {
+    entity(as[ReqOpenStake]) { body =>
+      buildOpenStakeTransaction(body) match {
+        case Success(tx) =>
+          if (body.automaticSend.getOrElse(true)) {
+            validateAndSendTransaction(tx.asInstanceOf[SidechainTypes#SCBT])
+          } else {
+            if (body.format.getOrElse(false))
+              ApiResponseUtil.toResponse(TransactionDTO(tx.asInstanceOf[SCBT]))
+            else
+              ApiResponseUtil.toResponse(TransactionBytesDTO(BytesUtils.toHexString(companion.toBytes(tx.asInstanceOf[SCBT]))))
+          }
+        case Failure(exception) =>
+          ApiResponseUtil.toResponse(GenericTransactionError("GenericTransactionError", JOptional.of(exception)))
+      }
+    }
+  }
+
+  def createOpenStakeTransactionSimplified: Route = (post & path("createOpenStakeTransactionSimplified")) {
+    entity(as[ReqOpenStakeSimplified]) { body =>
+     buildOpenStakeTransactionSimplified(body) match {
+       case Success(tx) =>
+         if (body.automaticSend.getOrElse(true)) {
+           validateAndSendTransaction(tx.asInstanceOf[SidechainTypes#SCBT])
+         } else {
+           if (body.format.getOrElse(false))
+             ApiResponseUtil.toResponse(TransactionDTO(tx.asInstanceOf[SCBT]))
+           else
+             ApiResponseUtil.toResponse(TransactionBytesDTO(BytesUtils.toHexString(companion.toBytes(tx.asInstanceOf[SCBT]))))
+         }
+       case Failure(exception) =>
+         ApiResponseUtil.toResponse(GenericTransactionError("GenericTransactionError", JOptional.of(exception)))     }
+    }
+  }
+
+  private def buildOpenStakeTransactionSimplified(body: ReqOpenStakeSimplified): Try[OpenStakeTransaction] = {
+    applyOnNodeView { sidechainNodeView =>
+      val wallet = sidechainNodeView.getNodeWallet
+
+      //Collect input box
+      val inputBox = wallet.allBoxes().asScala
+        .filter(box => box.proposition().equals(PublicKey25519PropositionSerializer.getSerializer.parseBytes(BytesUtils.fromHexString(body.forgerProposition))))
```

Comment:
`find` can be used instead of `filter`. Here we need to get the first one matched.
Moreover, extend the search rule with the `box.value >= fee` check.
Can be interesting to add this case check to the python test.

Reply:
Done